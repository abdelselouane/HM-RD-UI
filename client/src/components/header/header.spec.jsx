/* global beforeAll, describe, it, expect, jest */
import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import 'jest-styled-components';

import Header from './index';

describe('<Header />', () => {
  let wrapper;
  let props;
  let userProfile;

  beforeAll(() => {
    userProfile = {
      full_name: 'test user',
      id: 'testLDAP',
      groups: [{
        cn: 'Repair Depot Admin',
        dn: 'cn=Repair Depot Admin, ou=st9100, o=homedepot.com, c=US'
      }]
    };
    props = {
      showActions: false,
      handleLogoutClick: jest.fn(),
      handleLogoClick: jest.fn(),
      handleAdminClick: jest.fn(),
      userProfile: null
    };
    wrapper = mount(<Header {...props} />);
  });

  it('should render properly', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
    wrapper.setProps({ ...props, userProfile, showActions: true });
    expect(toJson(wrapper)).toMatchSnapshot();
    wrapper.setProps({ showAdmin: true });
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render Home Depot Logo as button if userProfile showActions', () => {
    wrapper.setProps({ ...props });
    const headerIcon = wrapper.find('div.header-logo');
    expect(headerIcon).toHaveLength(1);
    expect(headerIcon.prop('onClick')).not.toBeDefined();
    expect(wrapper.find('i.icon_homedepot')).toHaveLength(1);
  });

  it('should render product info', () => {
    expect(wrapper.find('.product-info').text()).toBe('Repair Depot');
  });

  it('should render username and logout if prop showActions is false', () => {
    wrapper.setProps({ ...props, userProfile, showActions: true });
    expect(wrapper.find('.header-actions .username').text()).toBe('testLDAP');
    expect(wrapper.find('.header-actions .logout-btn')).toHaveLength(1);
  });

  it('should call handleLogoutClick on logout-btn click', () => {
    const spy = jest.spyOn(props, 'handleLogoutClick');
    wrapper.find('.logout-btn').simulate('click');
    expect(spy).toHaveBeenCalled();
    spy.mockReset();
    spy.mockRestore();
  });

  it('should not render username and logout if prop showActions is false', () => {
    wrapper.setProps({ ...props, userProfile, showActions: false });
    expect(wrapper.find('.header-actions .username')).toHaveLength(0);
    expect(wrapper.find('.header-actions .logout-btn')).toHaveLength(0);
  });

  it('it should not show admin button when userprofile is null', () => {
    wrapper.setProps({ ...props, userProfile: null, showActions: false });
    expect(wrapper.find('.admin-btn')).toHaveLength(0);
  });

  it('it should have an admin button', () => {
    const spy = jest.spyOn(props, 'handleAdminClick');
    wrapper.setProps({ ...props, userProfile });
    expect(wrapper.find('.admin-btn')).toHaveLength(1);
    wrapper.find('.admin-btn').simulate('click');
    expect(spy).toHaveBeenCalled();
  });

  it('it should not show admin button when userprofile group is does not match ', () => {
    userProfile.groups = [];
    wrapper.setProps({ ...props, userProfile, showActions: false });
    expect(wrapper.find('.admin-btn')).toHaveLength(0);
  });
});
