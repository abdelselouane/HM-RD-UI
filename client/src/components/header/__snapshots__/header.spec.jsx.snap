// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should render properly 1`] = `
.c0 {
  z-index: 3;
}

.c0 .header-info .product-info {
  font-size: 1.5em;
}

.c0 .header-logo.go-to-homepage {
  cursor: pointer;
}

<Header
  handleAdminClick={[MockFunction]}
  handleLogoClick={[MockFunction]}
  handleLogoutClick={[MockFunction]}
  showActions={false}
  userProfile={null}
>
  <styled.div
    className="header"
  >
    <div
      className="header c0"
    >
      <div
        className="header-logo"
      >
        <i
          className="icon_homedepot"
        />
      </div>
      <div
        className="header-info"
      >
        <label
          className="product-info"
        >
          Repair Depot
        </label>
      </div>
      <div
        className="header-search"
      />
      <div
        className="header-actions"
      />
    </div>
  </styled.div>
</Header>
`;

exports[`<Header /> should render properly 2`] = `
.c0 {
  z-index: 3;
}

.c0 .header-info .product-info {
  font-size: 1.5em;
}

.c0 .header-logo.go-to-homepage {
  cursor: pointer;
}

<Header
  handleAdminClick={[MockFunction]}
  handleLogoClick={[MockFunction]}
  handleLogoutClick={[MockFunction]}
  showActions={true}
  userProfile={
    Object {
      "full_name": "test user",
      "groups": Array [
        Object {
          "cn": "Repair Depot Admin",
          "dn": "cn=Repair Depot Admin, ou=st9100, o=homedepot.com, c=US",
        },
      ],
      "id": "testLDAP",
    }
  }
>
  <styled.div
    className="header"
  >
    <div
      className="header c0"
    >
      <div
        className="header-logo go-to-homepage"
        onClick={[MockFunction]}
      >
        <i
          className="icon_homedepot"
        />
      </div>
      <div
        className="header-info"
      >
        <label
          className="product-info"
        >
          Repair Depot
        </label>
      </div>
      <div
        className="header-search"
      />
      <div
        className="header-actions"
      >
        <div
          className="admin-btn"
          onClick={[MockFunction]}
        >
          <i
            className="icon_laptop"
          />
          Admin
        </div>
        <div
          className="username"
        >
          <i
            className="icon_person"
          />
          testLDAP
        </div>
        <div
          className="logout-btn"
          onClick={[MockFunction]}
        >
          <i
            className="icon_exit_to_app"
          />
          Logout
        </div>
      </div>
    </div>
  </styled.div>
</Header>
`;

exports[`<Header /> should render properly 3`] = `
.c0 {
  z-index: 3;
}

.c0 .header-info .product-info {
  font-size: 1.5em;
}

.c0 .header-logo.go-to-homepage {
  cursor: pointer;
}

<Header
  handleAdminClick={[MockFunction]}
  handleLogoClick={[MockFunction]}
  handleLogoutClick={[MockFunction]}
  showActions={true}
  showAdmin={true}
  userProfile={
    Object {
      "full_name": "test user",
      "groups": Array [
        Object {
          "cn": "Repair Depot Admin",
          "dn": "cn=Repair Depot Admin, ou=st9100, o=homedepot.com, c=US",
        },
      ],
      "id": "testLDAP",
    }
  }
>
  <styled.div
    className="header"
  >
    <div
      className="header c0"
    >
      <div
        className="header-logo go-to-homepage"
        onClick={[MockFunction]}
      >
        <i
          className="icon_homedepot"
        />
      </div>
      <div
        className="header-info"
      >
        <label
          className="product-info"
        >
          Repair Depot
        </label>
      </div>
      <div
        className="header-search"
      />
      <div
        className="header-actions"
      >
        <div
          className="admin-btn"
          onClick={[MockFunction]}
        >
          <i
            className="icon_laptop"
          />
          Admin
        </div>
        <div
          className="username"
        >
          <i
            className="icon_person"
          />
          testLDAP
        </div>
        <div
          className="logout-btn"
          onClick={[MockFunction]}
        >
          <i
            className="icon_exit_to_app"
          />
          Logout
        </div>
      </div>
    </div>
  </styled.div>
</Header>
`;
